client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.ts

```diff
@@ -442,4 +214,18 @@ export class SDegreePlanEditorComponent implements OnInit {
       this.planService.removeCourse(indexes, this.plan.years);
     }
   }
+
+  /**
+   * Handles clicking on the "Submit" button
+   */
+  onClickSubmitPlan(): void {
```

Comment:
handles onClick events for the submit button. Actually calls the backend to submit new plan.